tests/unit/AppHelperTest.php

```diff
@@ -0,0 +1,49 @@
+<?php
+namespace unit;
+
+use uhi67\umvc\AppHelper;
+
+class AppHelperTest extends \Codeception\Test\Unit
+{
+    /**
+     * @var \UnitTester
+     */
+    protected $tester;
+    
+    protected function _before()
+    {
+    }
+
+    protected function _after()
+    {
+    }
+
+    // tests
+
+	/**
+	 * @dataProvider provWaitFor
+	 * @return void
+	 */
+    public function testWaitFor($timeout, $interval, $length, $success, $attempts, $elapsed)
+    {
+		$start = time();
+		$end = $start + $length;
+	    $a = 0;
+	    $result = AppHelper::waitFor(function() use($end, &$a) {
+			$a++;
+		    return time() >= $end;
+	    }, $timeout, $interval);
+
+		$this->assertEquals($success, $result);
+		$this->assertEquals($attempts, $a);
+		$this->assertEqualsWithDelta($elapsed, time()-$start, 1.0);
+
+    }
+	public function provWaitFor() {
+		return [
+			// $timeout, $interval, $length, $success, $attempts, $elapsed
+			[10, 1, 3, true, 4, 3],
+			[3, 2, 4, false, 2, 3],
+		];
+	}
+}
```

Comment:
Test output looks as follows. Maybe the expected wait duration can be displayed in the message: _Wait for 3 secs_ for example; not necessary.
```
+ AppHelperTest: Wait for | #0 (3.03s)
+ AppHelperTest: Wait for | #1 (3.03s)
```

The tests are also quite minimal, because early timeout (i.e. when `$timeout` is smaller than `$interval`) should be tested too, including when either duration is 0 (if supported by the `AppHelper::waitFor()` function).

Reply:
This message is generated by codeception from the name of the tested method :-)
Well, I completed with the cases of 0: no values < 1 ara allowed :-)

Reply:
Yes I realized the message was inferred from the test method name. I think PHPUnit behaves similarly and allows for named datasets in data providers. Of course, this was just a suggestion and not necessary at all. I ended up suggesting this because I changed the tests locally so that they fail, but it took 9 seconds before I got a feedback from the CLI and I couldn't figure out what the problem was (of course, I knew I introduced a problem...). So you can ignore my comments!